src/Compilers/Core/CodeAnalysisTest/AnalyzerConfigTests.cs

```diff
@@ -1151,6 +1151,59 @@ public void NestedAnalyzerOptionsWithRoot()
                 options);
         }
 
+        [Fact]
+        public void NestedAnalyzerOptionsWithOverrides()
+        {
+            var configs = ArrayBuilder<AnalyzerConfig>.GetInstance();
+            configs.Add(Parse(@"
+[*.cs]
+dotnet_diagnostic.cs000.some_key = a_val", "/.editorconfig"));
+            configs.Add(Parse(@"
+[test.*]
+dotnet_diagnostic.cs000.some_key = b_val", "/src/.editorconfig"));
+
+            var options = GetAnalyzerConfigOptions(
+                new[] { "/src/test.cs", "/src/test.vb", "/root.cs" },
+                configs);
+            configs.Free();
+
+            VerifyAnalyzerOptions(
+                new[] {
+                    new[] { ("dotnet_diagnostic.cs000.some_key", "b_val") },
+                    new[] { ("dotnet_diagnostic.cs000.some_key", "b_val") },
+                    new[] { ("dotnet_diagnostic.cs000.some_key", "a_val") }
+               },
+                options);
+        }
+
+        [Fact]
+        public void NestedAnalyzerOptionsWithSectionOverrides()
+        {
+            var configs = ArrayBuilder<AnalyzerConfig>.GetInstance();
+            configs.Add(Parse(@"
+[*.cs]
+dotnet_diagnostic.cs000.some_key = a_val", "/.editorconfig"));
```

Comment:
Could you also include an analyzer option override? I see we have some tests in this file for one or the other, but not both together.